### PR TITLE
Add comparisions for string

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: std/assembly/string.ts [hunk 1]
```diff
@@ -127,6 +127,67 @@ export class String {
...
+    ) < 0;
+  }
+
+  @operator("<=")
+  private static __lte(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = min<isize>(left.length, right.length);
```
Comment (dcodeIO) [inline std/assembly/string.ts]: Performs two casts to isize here. Might it make sense to use min<i32> and upcast once below where <usize> is expected?

Comment (MaxGraey) [reply std/assembly/string.ts]: Hmm, but `min<isize>` and `max<isize>` used everywhere. I prefer leave this changes for you if you think this makes sense.

Comment (MaxGraey) [reply std/assembly/string.ts]: For [example](https://github.com/MaxGraey/assemblyscript/blob/87e034aaeddc6584e11e3e9d01e028f00ddccc1a/std/assembly/string.ts#L202)

Comment (dcodeIO) [reply std/assembly/string.ts]: In your example, it is used because both arguments are already casted to isize, I believe.

Comment (MaxGraey) [reply std/assembly/string.ts]: Ok, I fixed this. Hmm, but nothing changes in wast file...

Comment (dcodeIO) [reply std/assembly/string.ts]: Yeah, this doesn't change anything because `isize` is just a 32-bit integer in WASM32. This is theoretical for WASM64.

Comment (MaxGraey) [reply std/assembly/string.ts]: Got it

File: std/assembly/string.ts [hunk 2]
```diff
@@ -127,6 +127,83 @@ export class String {
...
+
+  @operator(">=")
+  private static __gte(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = <usize>min<i32>(left.length, right.length);
+    if (length == 0) {
+      return left.charCodeAt(0) >= right.charCodeAt(0);
```
Comment (dcodeIO) [inline std/assembly/string.ts]: doesn't this result in an out of bounds read on the zero-length one?

Comment (MaxGraey) [reply std/assembly/string.ts]: No, because string with zero length actually is "\0"

Comment (dcodeIO) [reply std/assembly/string.ts]: Is it? Should be i32-length + u16 code units in memory, not zero-terminated.

File: std/assembly/string.ts [hunk 3]
```diff
@@ -127,6 +127,79 @@ export class String {
...
+  }
+
+  @operator(">")
+  private static __gt(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    if (!left.length)  return false;
+    if (!right.length) return true;
```
[the remark this answers is not in the document]
Comment (MaxGraey) [reply std/assembly/string.ts]: Done!

**Comment by dcodeIO** [comment]
Thanks :)

**Pull request merged**